src/contracts/actions/CappedTokenTransfers.sol

```diff
@@ -8,88 +8,91 @@ import {ICappedTokenTransfers} from 'interfaces/actions/ICappedTokenTransfers.so
 import {IERC20} from 'forge-std/interfaces/IERC20.sol';
 
 contract CappedTokenTransfers is SafeManageable, ICappedTokenTransfers {
-  mapping(address _token => uint256 _transferCap) public tokenCap;
-  mapping(address _token => uint256 _transferCapSpent) public capSpent;
-  mapping(address _token => uint256 _transferCooldown) public tokenCooldown;
+  // Token configuration
+  address public immutable TOKEN;
+  uint256 public immutable CAP;
+  uint256 public immutable EPOCH_LENGTH;
+
+  // State tracking
+  uint256 public totalSpent;
+  uint256 public startingTimestamp;
 
   TokenTransfer[] public tokenTransfers;
 
-  constructor(address _safe) SafeManageable(_safe) {}
+  constructor(address _safe, address _token, uint256 _cap, uint256 _epochLength) SafeManageable(_safe) {
+    TOKEN = _token;
+    CAP = _cap;
+    EPOCH_LENGTH = _epochLength;
+    startingTimestamp = block.timestamp;
+  }
 
   // ~~~ ADMIN METHODS ~~~
 
-  function addCappedToken(address _token, uint256 _cap) external isSafe {
-    tokenCap[_token] = _cap;
+  function addTokenTransfer(address _recipient, uint256 _amount) external isSafeOwner {
+    tokenTransfers.push(TokenTransfer({recipient: _recipient, amount: _amount}));
   }
 
-  function addTokenTransfer(address _token, address _recipient, uint256 _amount) external isSafeOwner {
-    _addTokenTransfer(_token, _recipient, _amount);
-  }
+  function removeTokenTransfer(uint256 _index) external isSafeOwner {
+    if (_index >= tokenTransfers.length) revert InvalidIndex();
 
-  function addTokenTransfers(
-    address[] memory _tokens,
-    address[] memory _recipients,
-    uint256[] memory _amounts
-  ) external isSafeOwner {
-    if (_tokens.length != _recipients.length || _tokens.length != _amounts.length) {
-      revert LengthMismatch();
-    }
-    for (uint256 i = 0; i < _tokens.length; i++) {
-      _addTokenTransfer(_tokens[i], _recipients[i], _amounts[i]);
-    }
+    delete tokenTransfers[_index];
   }
 
   // ~~~ ACTIONS METHODS ~~~
 
-  function getActions() external returns (Action[] memory) {
-    Action[] memory actions = new Action[](tokenTransfers.length);
-
+  function getActions() external view returns (Action[] memory) {
+    // Count valid transfers
+    uint256 _validCount = 0;
+    uint256 _totalAmount = 0;
     for (uint256 i = 0; i < tokenTransfers.length; i++) {
-      TokenTransfer memory tokenTransfer = tokenTransfers[i];
-      actions[i] = Action({
-        target: tokenTransfer.token,
-        data: abi.encodeWithSelector(IERC20.transfer.selector, tokenTransfer.recipient, tokenTransfer.amount),
-        value: 0
-      });
-      capSpent[tokenTransfer.token] += tokenTransfer.amount;
+      if (tokenTransfers[i].amount != 0) {
+        _validCount++;
+        _totalAmount += tokenTransfers[i].amount;
+      }
```

Comment:
hmm ok, i guess it gets handled here, we should add a zero-check in the `addTokenTransfer` method